pkg/manager/member/ticdc_scaler.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package member
+
+import (
+	"fmt"
+
+	apps "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog"
+
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
+	"github.com/pingcap/tidb-operator/pkg/controller"
+	"github.com/pingcap/tidb-operator/pkg/util"
+)
+
+type ticdcScaler struct {
+	generalScaler
+}
+
+// NewTiCDCScaler returns a TiCDC Scaler.
+func NewTiCDCScaler(deps *controller.Dependencies) *ticdcScaler {
+	return &ticdcScaler{generalScaler: generalScaler{deps: deps}}
+}
+
+// Scale scales in or out of the statefulset.
+func (s *ticdcScaler) Scale(meta metav1.Object, oldSet *apps.StatefulSet, newSet *apps.StatefulSet) error {
+	scaling, _, _, _ := scaleOne(oldSet, newSet)
+	if scaling > 0 {
+		return s.ScaleOut(meta, oldSet, newSet)
+	} else if scaling < 0 {
+		return s.ScaleIn(meta, oldSet, newSet)
+	}
+	// we only sync auto scaler annotations when we are finishing syncing scaling
```

Comment:
Remove this line.

Reply:
https://github.com/pingcap/tidb-operator/pull/4004/commits/472f2e93cc0c6be310c36d8ba78c049cdf023190